zrml/swaps/src/lib.rs

```diff
@@ -1377,6 +1375,7 @@ mod pallet {
                     let mut account_created = false;
                     let mut total_balance = <BalanceOf<T>>::zero();
                     total_assets = pool.assets.len();
+                    let mut providers_and_pool_shares = vec![];
```

Comment:
Not sure if that is true but we probably can save some time that would be used for reallocations by specifying the capacity of the vector `with_capacity` based on the `decode_len` of the `<SubsidyProviders<T>>::decode_len(pool_id)`.

My assumption is that for small values it's slower, but for big values it faster, but where is the border? We definitely lose about 25µs for the extra storage read to get the length beforehand.

Reply:
Hmm... When there's time, I guess we could benchmark this.

Ok, maybe another thing that's related here: I was thinking if there should be a maximum number of subsidy providers, let's say 100. In that case, we could just go with that number as capacity.

Reply:
That idea is actually not that bad. We'd need a far bigger number to avoid users creating multiple accounts and clogging the list with minimal amounts.
I think we can keep that in mind for the future.

Reply:
In addition to that, the number must be reasonably low such that it won't cause overweight blocks that halt the chain.

Reply:
Yes. I guess we could combine it with a minimum amount of subsidy per provider.